Ryujinx.HLE/HOS/Services/Sockets/Bsd/IClient.cs

```
@@ -530,7 +530,12 @@ public ResultCode RecvFrom(ServiceCtx context)
 
                     receiveRegion.Dispose();
 
-                    context.Memory.Write(sockAddrOutPosition, BsdSockAddr.FromIPEndPoint(endPoint));
+                    var bsdSockAddr = BsdSockAddr.FromIPEndPoint(endPoint);
+
+                    if (sockAddrOutSize != 0 && sockAddrOutSize >= (ulong) Unsafe.SizeOf<BsdSockAddr>())
```

You should probably return an error if we can't write the address to memory.

---

Do you suggest to change the errno which is LinuxError enum? if so what should be the error?

---

`LinuxError.ENOMEM` sounds fitting, but I should add that I currently don't know if that's the way HOS deals with it.